src/sentry/insights/migrations/0002_backfill_team_starred.py

```diff
@@ -0,0 +1,97 @@
+# Generated by Django 5.2.1 on 2025-06-10 16:25
+import logging
+from typing import cast
+
+from django.db import IntegrityError, migrations, router, transaction
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+from sentry.discover.models import TeamKeyTransaction
+from sentry.insights.models import InsightsStarredSegment
+from sentry.models.organization import Organization
+from sentry.models.organizationmemberteam import (
+    OrganizationMemberTeam as OrganizationMemberTeamType,
+)
+from sentry.models.projectteam import ProjectTeam
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+logger = logging.getLogger(__name__)
+
+
+def get_user_ids_from_team(projectTeam: ProjectTeam, apps: StateApps):
+    OrganizationMemberTeam = cast(
+        OrganizationMemberTeamType, apps.get_model("sentry", "Organizationmemberteam")
+    )
+
+    team = projectTeam.team
+    team_members = OrganizationMemberTeam.objects.filter(team=team)
+    return [member.organizationmember.user_id for member in team_members]
+
+
+def migrate_team_stars_to_user_stars(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor):
+    TeamStarredTransaction = cast(
+        TeamKeyTransaction, apps.get_model("sentry", "teamkeytransaction")
+    )
+    queryset = TeamStarredTransaction.objects.filter()
+    for team_starred_transaction in RangeQuerySetWrapperWithProgressBar(queryset):
+        starred_transaction: TeamKeyTransaction = team_starred_transaction
+
+        segment_name: str = starred_transaction.transaction
+        projectTeam: ProjectTeam = starred_transaction.project_team
+        organization: Organization = starred_transaction.organization
+        project = projectTeam.project
+
+        user_ids = get_user_ids_from_team(projectTeam, apps)
+        segments_to_create = [
+            InsightsStarredSegment(
+                organization_id=organization.id,
+                project_id=project.id,
```

Comment:
Since you don't actually use the project here, you can just use `projectTeam.project_id` instead and skip fetching the project. Same for `organization.id`